src/create3-fastslam.cpp

```diff
@@ -96,73 +92,68 @@ class FastSLAMC3: public rclcpp::Node
   rclcpp::Subscription<sensor_msgs::msg::LaserScan>::SharedPtr m_landmark_sub;
   void lm_callback(const sensor_msgs::msg::LaserScan::SharedPtr msg);
   rclcpp::Publisher<visualization_msgs::msg::Marker>::SharedPtr m_observation_visualization_pub;
-  rclcpp::Publisher<nav_msgs::msg::Odometry>::SharedPtr m_deadreckon_odom;
-  void publishSLAMOdom();
-
+#ifdef VISUALIZE_ROB_PATH 
+  std::vector<geometry_msgs::msg::Point> m_path_points; //for path visualization
+  rclcpp::Publisher<visualization_msgs::msg::Marker>::SharedPtr m_path_visualization_pub;
+  void visualizeSLAMOdom();
+#endif
   //FastSLAM member variables
   std::unique_ptr<FastSLAMPF> m_fastslam_filter;
   std::shared_ptr<Create3Manager> m_robot_manager;
 
   //robot parameters
-  struct Pose2D m_rob_pose_delta;
-  bool m_pose_calibrated; // discard first sample
```

Comment:
Did you eventually figure out how to reset the odometry through this node? I guess we can manually reset it every time before we run this but it would be nice if there is an automatic way of resetting pose, whether through a separate process or maybe through the launch file.

Reply:
I added it to the launch file, so we no longer have to worry about it